doku/src/objects/type.rs

```diff
@@ -35,3 +35,263 @@ impl From<TypeKind> for Type {
         }
     }
 }
+
+impl From<Fields> for Type {
+    fn from(fields: Fields) -> Self {
+        let kind: TypeKind = fields.into();
+        kind.into()
+    }
+}
+
+impl Type {
+    /// Recursively walks through the current type, and when an enum is found,
+    /// all its variants are converted to equivalent structs.
+    pub fn enum_to_structs(&self) -> Option<Vec<Self>> {
```

Comment:
Could we make it `pub(crate)`?

Having helper-structs is handy, but I'm not sure if we should make them public just yet.